modyn/trainer_server/internal/trainer/pytorch_trainer.py

```diff
@@ -294,15 +317,18 @@ def train(self) -> None:  # pylint: disable=too-many-locals, too-many-branches
                 pre_downsampling_size = target.shape[0]
 
                 with torch.autocast(self._device_type, enabled=self._amp):
-                    if self._downsampling_enabled:
+                    if self._downsampling_mode == DownsamplingMode.BATCH_THEN_SAMPLE:
                         # TODO(#218) Persist information on the sample IDs/weights when downsampling is performed
                         assert self._downsampler is not None
+                        self._downsampler.init_downsampler()
```

Comment:
so init is also like a reset function? because we use the same downsampler object multiple times, right?

Reply:
Yes, it's also a reset function (basically it is called once in STB and every batch in BTS)